### ISSUE TITLE
contextaul attention normalization

### ISSUE THREAD

**Issue by powder21** [JiahuiYu/generative_inpainting#274]
Why do you normalize background patch by wi_normed = wi / tf.maximum(tf.sqrt(tf.reduce_sum(tf.square(wi), axis=[0,1,2])), 1e-4), but not normalize foreground. (Cosine similarity mentioned in your paper both normalizes these two kinds of patches.)

**Comment by JiahuiYu** [comment]
Normalizing foreground will be canceled if you look over entire formulation.

**Comment by powder21** [comment]
> Normalizing foreground will be canceled if you look over entire formulation.

`yi = tf.nn.conv2d(xi, wi_normed, strides=[1,1,1,1], padding="SAME")
yi = tf.nn.softmax(yi*scale, 3)
yi = tf.nn.conv2d_transpose(yi, wi_center, tf.concat([[1], raw_fs[1:]], axis=0), strides=[1,rate,rate,1]) / 4.
`
The foreground patch are used in these lines, but I don't understand where normalization is canceled. Could u explain it in detail?
